request.go

```diff
@@ -71,6 +72,59 @@ type Request struct {
 	StatusCode int               `json:"statuscode,omitempty"`
 }
 
+type RequestListFilter struct {
+	url.Values
+}
+
+func NewRequestListFilter() *RequestListFilter {
+	return &RequestListFilter{Values: url.Values{}}
+}
+
+func (f *RequestListFilter) AddUrl(url string) {
+	f.WithUrl(url)
+}
+
+func (f *RequestListFilter) WithUrl(url string) *RequestListFilter {
+	f.Add("filter.url", url)
+	return f
+}
+
+func (f *RequestListFilter) AddCreatedDate(createdDate string) {
+	f.WithCreatedDate(createdDate)
+}
+
+func (f *RequestListFilter) WithCreatedDate(createdDate string) *RequestListFilter {
+	f.Add("filter.createdDate", createdDate)
+	return f
+}
+
+func (f *RequestListFilter) AddMethod(method string) {
+	f.WithMethod(method)
+}
+
+func (f *RequestListFilter) WithMethod(method string) *RequestListFilter {
+	f.Add("filter.method", method)
+	return f
+}
+
+func (f *RequestListFilter) AddBody(body string) {
+	f.WithBody(body)
+}
+
+func (f *RequestListFilter) WithBody(body string) *RequestListFilter {
+	f.Add("filter.body", body)
+	return f
+}
+
+func (f *RequestListFilter) AddRequestStatus(requestStatus string) {
```

Comment:
This one surprises me, as for the `Request` struct the RequestStatus` is another struct so what will the given string actually be matched against?

Reply:
Request.Metadata.RequestStatus.Metadata.Status

Reply:
Other filters where tested by hand as they are not documented

Reply:
which other filters? all the others were documented, just this one was unclear

Reply:
body and url where not documented too. So 3/5

Reply:
They both fall into the category described here: https://devops.ionos.com/api/cloud/v4/#custom-query-response-filters

Reply:
I see.